### PR TITLE
Adapt leader-election-api to baseline requirements

### PR DESCRIPTION
**Pull request by fsamuel-bs**
**Goals (and why)**: Adapt the leader-election-api module to baseline requirements.

**Implementation Description (bullets)**: -

**Concerns (what feedback would you like?)**: What should I do in the "REVIEW" tags around the code.

**Where should we start reviewing?**: -

**Priority (whenever / two weeks / yesterday)**: EOD

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1856)

### CODE REVIEW

File: leader-election-api/src/main/java/com/palantir/leader/PingableLeader.java [hunk 1]
```diff
@@ -34,10 +34,11 @@
     boolean ping();
 
     /**
-     * @return a unique string identifier for the leader election service
+     * Returns a unique string identifier for the leader election service
      */
     @GET
     @Path("uuid")
     @Produces(MediaType.TEXT_PLAIN)
+    // REVIEW.
```
Comment (fsamuel-bs) [inline leader-election-api/src/main/java/com/palantir/leader/PingableLeader.java]: What should I do with UUID? Rename everything to Uuid?

File: leader-election-api/src/main/java/com/palantir/paxos/PaxosPromise.java [hunk 2]
```diff
@@ -28,6 +28,7 @@
 /**
  * A promise to not accept new proposals less than promisedID.
  *
+ * REVIEW:
```
Comment (fsamuel-bs) [inline leader-election-api/src/main/java/com/palantir/paxos/PaxosPromise.java]: Should I delete the line below?

File: leader-election-api/src/main/java/com/palantir/paxos/PaxosPromise.java [hunk 3]
```diff
@@ -66,7 +66,7 @@ public static PaxosPromise create(
 
     private PaxosPromise(PaxosProposalId promisedId) {
         ack = false;
-        this.promisedId = Preconditions.checkNotNull(promisedId);
+        this.promisedId = Preconditions.checkNotNull(promisedId, "promiseId cannot be null");
```
Comment (gsheasby) [inline leader-election-api/src/main/java/com/palantir/paxos/PaxosPromise.java]: Typo: `promisedId` (ditto below)

File: leader-election-impl/src/main/java/com/palantir/paxos/PaxosProposerImpl.java [hunk 4]
```diff
@@ -203,7 +203,7 @@ public int getQuorumSize() {
     }
 
     @Override
-    public String getUUID() {
+    public String getUuid() {
```
Comment (jboreiko) [inline leader-election-impl/src/main/java/com/palantir/paxos/PaxosProposerImpl.java]: This is a breaking change.

Comment (gsheasby) [inline leader-election-impl/src/main/java/com/palantir/paxos/PaxosProposerImpl.java]: I undid a number of these changes in https://github.com/palantir/atlasdb/pull/1856/commits/b1e4141b4968442694efe0a0441e32e2a341f72e - does this one actually break any external products?

Comment (jboreiko) [inline leader-election-impl/src/main/java/com/palantir/paxos/PaxosProposerImpl.java]: ah, cool.  Not sure, just wanted us to be conscious of it and I didn't see any discussion about how it was breaking.

**Pull request merged**